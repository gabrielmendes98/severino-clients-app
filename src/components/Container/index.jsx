import React, { useMemo } from 'react';
import { View } from 'react-native';
import PropTypes from 'prop-types';
import createStyles from './style';

const Container = ({ children, style, horizontalSpacing }) => {
  const styles = useMemo(
    () => createStyles({ horizontalSpacing }),
    [horizontalSpacing],
  );

  return <View style={[styles.container, style]}>{children}</View>;
};

Container.defaultProps = {
  horizontalSpacing: true,
};

Container.propTypes = {
  children: PropTypes.any,
  style: PropTypes.object,
  horizontalSpacing: PropTypes.bool,
};

export default Container;
