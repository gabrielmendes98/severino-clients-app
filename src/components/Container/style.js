import { StyleSheet } from 'react-native';
import { SAFE_SPACING } from 'common/constants';
import theme from 'common/styles/theme';

const createStyles = ({ horizontalSpacing, removeBottomMargin }) =>
  StyleSheet.create({
    container: {
      flex: 1,
      marginTop: theme.spacing(SAFE_SPACING),
      paddingHorizontal: horizontalSpacing ? theme.spacing(SAFE_SPACING) : 0,
      marginBottom: removeBottomMargin ? 0 : theme.spacing(SAFE_SPACING),
    },
  });

export default createStyles;
