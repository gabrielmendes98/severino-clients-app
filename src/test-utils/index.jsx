/* eslint-disable react/prop-types */
import React from 'react';
import { render } from '@testing-library/react-native';
import MainProvider from 'common/contexts/Main';

const Wrapper = ({ children }) => <MainProvider>{children}</MainProvider>;

const customRender = (ui, options) =>
  render(ui, { wrapper: Wrapper, ...options });

export * from '@testing-library/react-native';
export { customRender as render };
