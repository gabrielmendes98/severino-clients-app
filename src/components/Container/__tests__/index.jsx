import React from 'react';
import { View } from 'react-native';
import { rawRender } from 'test-utils';
import Container from '../index';

it('should renders correctly when pass styling props as true', () => {
  const { toJSON } = rawRender(
    <Container horizontalSpacing removeBottomMargin>
      <View>TEST</View>
    </Container>,
  );
  expect(toJSON()).toMatchSnapshot();
});

it('should renders correctly when pass styling props as false', () => {
  const { toJSON } = rawRender(
    <Container horizontalSpacing={false} removeBottomMargin={false}>
      <View>TEST</View>
    </Container>,
  );
  expect(toJSON()).toMatchSnapshot();
});
