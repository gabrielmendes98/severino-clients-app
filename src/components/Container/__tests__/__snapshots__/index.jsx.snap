// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should renders correctly when pass styling props as false 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "marginBottom": 12,
        "marginTop": 12,
        "paddingHorizontal": 0,
      },
      undefined,
    ]
  }
>
  <View>
    TEST
  </View>
</View>
`;

exports[`should renders correctly when pass styling props as true 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "marginBottom": 0,
        "marginTop": 12,
        "paddingHorizontal": 12,
      },
      undefined,
    ]
  }
>
  <View>
    TEST
  </View>
</View>
`;
